### ISSUE TITLE
script tags emptied by safe_html transform

### ISSUE THREAD

**Issue by frisi** [plone/Products.CMFPlone#1230]
if you want your editors to be able to add `<script>` tags to the richtext field you're currently out of luck.
plone's safe_html transform will strip all of the script-tag's content.

#### motivation:

add [facebook page plugin](https://developers.facebook.com/docs/plugins/page-plugin) to a document:
```html
<div id="fb-root"></div>
<script>(function(d, s, id) {
  var js, fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) return;
  js = d.createElement(s); js.id = id;
  js.src = "//connect.facebook.net/de_DE/sdk.js#xfbml=1&version=v2.5";
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>

<div class="fb-page" data-href="https://www.facebook.com/facebook" data-small-header="false" data-adapt-container-width="true" data-hide-cover="false" data-show-facepile="true" data-show-posts="true"><div class="fb-xfbml-parse-ignore"><blockquote cite="https://www.facebook.com/facebook"><a href="https://www.facebook.com/facebook">Facebook</a></blockquote></div></div>
```

#### problem description

remove "script" from `nasty_tags` and `stripped_tags` in `@@filter-controlpanel`

tiny now allows to add the code via the "html" button, but wraps javascript with CDATA (so it's [not executed within tiny](http://www.tinymce.com/forum/viewtopic.php?id=29176))

```html
<script type="text/javascript">// <![CDATA[
(function(d, s, id) {
...
}(document, 'script', 'facebook-jssdk'));
// ]]></script>
```

safe_html then, strips the ``<![CDATA`` element leading to the following script tag:
```html
<script type="text/javascript">// </script>
```

according to [this discussion from 2012](http://plone.293351.n2.nabble.com/Problem-with-tiny-and-script-tag-td7561205.html) this has been with PloneHotfix20121106

#### possible solutions

when saving the text to the richtext field, remove the CDATA  wrapper (as suggested in [the tiny forum](http://www.tinymce.com/forum/viewtopic.php?id=29176))
or make safe_html not strip the CDATA

using plone 4.3.7

**Comment by sneridagh** [comment]
Just tried in 4.3.9 and works fine. As stated here:
http://glicksoftware.com/blog/disable-html-filtering

For the record: in Plone 5 is required to add:
{"extended_valid_elements": "script[language|type|src]"}

in TinyMCE settings -> Advanced -> Other settings to enable it.

**Comment by davilima6** [comment]
This "Other settings" step need to be documented here: http://docs.plone.org/adapt-and-extend/config/html-filtering.html#plone-5

And/or here: http://docs.plone.org/develop/plone/forms/wysiwyg.html?highlight=tinymce#disabling-html-filtering-and-safe-html-transformation

Btw shouldn't Tiny grab that from @@filter-controlpanel's `custom_tags`?

**Comment by sneridagh** [comment]
@davilima6 Indeed, they should be documented at least. Since I've been working on that today I will add it myself.

**Comment by jensens** [comment]
Closing as outdated (10 years old). About Plone 4.3.7 CDATA wrapping in script tags with TinyMCE 3/4. Plone 6 uses TinyMCE 6 which handles script tags differently.
If you think this issue is still relevant please reopen it. Thank you for the original report!
